### DISCUSSION TITLE
SMAA weights not working as expected

### DISCUSSION THREAD

**Discussion by AdrianFahrbach** [pmndrs/postprocessing#663] [Q&A] [answered]
Hey there!

I'm currently trying to improve the antialiasing of a webgl Three.js app. We are using the `SMAAEffect`.
I've tried playing around with the settings a bit by copying the tweak panel from [the demo](https://github.com/pmndrs/postprocessing/blob/main/manual/assets/js/src/demos/smaa.js#L101), but no matter what I try, the effect of SMAA is always very little to not visible at all.

To me, it looks like my edge detection is fine, but the weighted result that I get are "broken". If I compare my results to what I see in the [interactive demo](https://pmndrs.github.io/postprocessing/public/demo/#antialiasing), then it feels like I'm doing something wrong.

Do you have an idea what I could be missing? Or is this maybe even expected, since my scene is quite dark and my edges are relatively small.
My Three.js knowledge is quite limited, so any input is highly appreciated!

![vlcsnap-2024-10-11-23h06m47s083](https://github.com/user-attachments/assets/a2f124d4-5925-4180-9fde-e58f969ae61d)
![vlcsnap-2024-10-11-23h08m21s889](https://github.com/user-attachments/assets/ea2f79d3-5293-44fc-bc7e-a2ca08812386)
![vlcsnap-2024-10-11-23h08m43s810](https://github.com/user-attachments/assets/c1b1d2df-6c10-41fd-948e-ca67345af9b2)

https://github.com/user-attachments/assets/2d95cbf8-8428-40ff-91b9-799bb2a876ad

**Comment by vanruesc** [comment, marked as answer]
Hi, the weights seem to be calculated correctly and the detected edges seem to be smoothed as expected.

If you're talking about the broken lines and shimmering in motion, then that's not something SMAA can fix. You'd need to use MSAA to get more stable highlights in a dark scene like this. TAA would likely also yield better results in this case, but that's currently not included in `postprocessing`.

For SMAA I'd recommend sticking to the color edge detection with depth-based predication. Use a low `predicationThreshold` around `0.0002`, a `predicationStrength` of `1` and a `predicationScale` of `2`. Also, make sure to optimize your depth buffer range by setting the camera's `near` plane to something like `0.3` or `1.0`. This increases the accuracy of the predication. The `far` setting doesn't matter much for this.

**Reply by AdrianFahrbach** [reply]
Makes sense. I switched to MSAA and it works perfectly.
Thank you so much! Awesome to see that you are always responding to the discussions here with such a technical depth!
